<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>899</width>
    <height>704</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_6">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="10,4">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5" stretch="10,3">
        <item>
         <widget class="QTextEdit" name="textEdit">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPlainTextEdit" name="sendTextEdit"/>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,1">
            <property name="spacing">
             <number>4</number>
            </property>
            <item>
             <widget class="QCheckBox" name="addNewlineChkBox">
              <property name="minimumSize">
               <size>
                <width>76</width>
                <height>18</height>
               </size>
              </property>
              <property name="text">
               <string>添加换行</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="clearBtn">
              <property name="minimumSize">
               <size>
                <width>76</width>
                <height>22</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">color: red;</string>
              </property>
              <property name="text">
               <string>清空</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="sendBtn">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize">
               <size>
                <width>76</width>
                <height>24</height>
               </size>
              </property>
              <property name="styleSheet">
               <string notr="true">color: green;</string>
              </property>
              <property name="text">
               <string>发送</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QScrollArea" name="scrollArea">
        <property name="minimumSize">
         <size>
          <width>200</width>
          <height>0</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>260</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>249</width>
           <height>643</height>
          </rect>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="leftMargin">
           <number>6</number>
          </property>
          <property name="topMargin">
           <number>8</number>
          </property>
          <property name="rightMargin">
           <number>6</number>
          </property>
          <property name="bottomMargin">
           <number>8</number>
          </property>
          <item>
           <widget class="QGroupBox" name="portGroupBox">
            <property name="title">
             <string>串口设置</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_7">
             <property name="leftMargin">
              <number>4</number>
             </property>
             <property name="topMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>4</number>
             </property>
             <property name="bottomMargin">
              <number>6</number>
             </property>
             <item>
              <layout class="QFormLayout" name="formLayout">
               <item row="0" column="1">
                <widget class="QComboBox" name="portComboBox"/>
               </item>
               <item row="1" column="0">
                <widget class="QLabel" name="baudLabel">
                 <property name="text">
                  <string>波特率：</string>
                 </property>
                </widget>
               </item>
               <item row="1" column="1">
                <widget class="QComboBox" name="baudComboBox">
                 <property name="editable">
                  <bool>true</bool>
                 </property>
                 <item>
                  <property name="text">
                   <string notr="true">4800</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">9600</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">14400</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">19200</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">38400</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">56000</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">57600</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">115200</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">128000</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">256000</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">460800</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">512000</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">750000</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">921600</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">1500000</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="2" column="0">
                <widget class="QLabel" name="databitsLabel">
                 <property name="text">
                  <string>数据位：</string>
                 </property>
                </widget>
               </item>
               <item row="2" column="1">
                <widget class="QComboBox" name="databitsComboBox">
                 <item>
                  <property name="text">
                   <string notr="true">5</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">6</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">7</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string notr="true">8</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="3" column="0">
                <widget class="QLabel" name="parityLabel">
                 <property name="text">
                  <string>校验位：</string>
                 </property>
                </widget>
               </item>
               <item row="3" column="1">
                <widget class="QComboBox" name="parityComboBox">
                 <item>
                  <property name="text">
                   <string>NONE</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>ODD</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>EVEN</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>MARK</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>SPACE</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="4" column="0">
                <widget class="QLabel" name="stopbitsLabel">
                 <property name="text">
                  <string>停止位：</string>
                 </property>
                </widget>
               </item>
               <item row="4" column="1">
                <widget class="QComboBox" name="stopbitsComboBox">
                 <item>
                  <property name="text">
                   <string>1</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>1.5</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>2</string>
                  </property>
                 </item>
                </widget>
               </item>
               <item row="0" column="0">
                <widget class="QLabel" name="portLabel">
                 <property name="text">
                  <string>端口号：</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="0">
                <widget class="QLabel" name="flowCtrlLabel">
                 <property name="text">
                  <string>流控制</string>
                 </property>
                </widget>
               </item>
               <item row="5" column="1">
                <widget class="QComboBox" name="flowCtrlComboBox">
                 <item>
                  <property name="text">
                   <string>NONE</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>XON/XOFF</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>RTS/CTS</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>DTR/DSR</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>RTS/CTS/XON/XOFF</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>DTR/DSR/XON/XOFF</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="openPortBtn">
               <property name="text">
                <string>打开串口</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="recvGroupBox">
            <property name="title">
             <string>接收设置</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_8">
             <property name="leftMargin">
              <number>4</number>
             </property>
             <property name="topMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>4</number>
             </property>
             <property name="bottomMargin">
              <number>6</number>
             </property>
             <item>
              <widget class="QCheckBox" name="hexShowChkBox">
               <property name="text">
                <string>以十六进制显示</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="autoSaveToFileChkBox">
               <property name="text">
                <string>数据自动保存到文件</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="5,1">
               <property name="spacing">
                <number>4</number>
               </property>
               <item>
                <widget class="QLineEdit" name="saveFilePathEdit">
                 <property name="placeholderText">
                  <string>设置文件路径</string>
                 </property>
                 <property name="clearButtonEnabled">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="saveFilePathBtn">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>...</string>
                 </property>
                 <property name="flat">
                  <bool>false</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QCheckBox" name="pauseShowRecvChkBox">
               <property name="text">
                <string>暂停显示接收数据</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QPushButton" name="saveDataBtn">
                 <property name="text">
                  <string>保存数据</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="clearDataBtn">
                 <property name="text">
                  <string>清空数据</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="sendGroupBox">
            <property name="title">
             <string>发送设置</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <property name="leftMargin">
              <number>4</number>
             </property>
             <property name="topMargin">
              <number>6</number>
             </property>
             <property name="rightMargin">
              <number>4</number>
             </property>
             <property name="bottomMargin">
              <number>6</number>
             </property>
             <item>
              <widget class="QCheckBox" name="hexSendChkBox">
               <property name="text">
                <string>以十六进制发送</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="sendFileChkBox">
               <property name="text">
                <string>发送文件</string>
               </property>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_8" stretch="5,1">
               <property name="spacing">
                <number>4</number>
               </property>
               <item>
                <widget class="QLineEdit" name="sendFilePathEdit">
                 <property name="placeholderText">
                  <string>设置文件路径</string>
                 </property>
                 <property name="clearButtonEnabled">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QPushButton" name="sendFilePathBtn">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Ignored" vsizetype="Fixed">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="text">
                  <string>...</string>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <property name="spacing">
                <number>2</number>
               </property>
               <item>
                <widget class="QCheckBox" name="timingSendChkBox">
                 <property name="text">
                  <string>定时发送</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QDoubleSpinBox" name="timerSpinBox">
                 <property name="alignment">
                  <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
                 </property>
                 <property name="buttonSymbols">
                  <enum>QAbstractSpinBox::NoButtons</enum>
                 </property>
                 <property name="decimals">
                  <number>3</number>
                 </property>
                 <property name="minimum">
                  <double>0.001000000000000</double>
                 </property>
                 <property name="maximum">
                  <double>9999.998999999999796</double>
                 </property>
                 <property name="singleStep">
                  <double>0.500000000000000</double>
                 </property>
                 <property name="value">
                  <double>1.000000000000000</double>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="sizePolicy">
                  <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
                   <horstretch>0</horstretch>
                   <verstretch>0</verstretch>
                  </sizepolicy>
                 </property>
                 <property name="maximumSize">
                  <size>
                   <width>22</width>
                   <height>22</height>
                  </size>
                 </property>
                 <property name="text">
                  <string>秒</string>
                 </property>
                 <property name="alignment">
                  <set>Qt::AlignCenter</set>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QCheckBox" name="showSentDataChkBox">
               <property name="text">
                <string>显示发送字符串</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="Line" name="line_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_6" stretch="2,3">
            <property name="spacing">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>换行符</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="newlineComboBox">
              <item>
               <property name="text">
                <string>\r\n (CRLF)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>\r (CR)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>\n (LF)</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_7" stretch="2,3">
            <property name="spacing">
             <number>2</number>
            </property>
            <item>
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>窗口主题</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="themeComboBox">
              <item>
               <property name="text">
                <string>白底黑字</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>白底绿字</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>白底蓝字</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>黑底白字</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>黑底绿字</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>黑底黄字</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>899</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
